docs/src/reference/message-correlation.md

```diff
@@ -1,35 +1,43 @@
+An instance of an order workflow has been created with the orderId variable set as `0000-123`, and is waiting at a message catch event until it receives a message to let it know that the money has been collected. The message catch event specifies the message name `Money collected` and the correlation key `orderId`.
 
-A message is published by the payment service using one of the Zeebe clients. It has the name `Money collected` and the correlation key `order-123`. Since the correlation information matches, the message is correlated to the workflow instance. That means its payload is merged into the workflow instance payload and the message catch event is left.
+A message is published by the payment service. It has the name `Money collected` and the message correlationKey is set as `0000-123`. The broker will correlate this message with any running workflows waiting on a message event with a message name `Money Collected`, whose orderId is `0000-123`. When the correlation information matches, the message is correlated to the workflow instance. That means that our order workflow instance will receive a copy of the message. The message payload is merged into the order workflow instance payload and the message catch event is left.
 
-> Note that a message can be correlated to multiple workflow instances if they share the same correlation information. But it can be correlated only once per workflow instance.
+> Note that a message can be correlated to multiple workflow instances if they share the same correlation information. But it will be correlated only once per workflow instance.
 
 ## Message Buffering
 
 In Zeebe, messages can be buffered for a given time. Buffering can be useful in a situation when it is not guaranteed that the message catch event is entered before the message is published.
 
-A message has a time-to-live (TTL) which specifies the time how long it is buffered. Within this time, the message can be correlated to a workflow instance.
+A message has a time-to-live (TTL) which specifies the time how long it is buffered. Within this time, the message can be correlated to any number of workflow instances where there is a match.
+
+When a workflow instance enters a message catch event, it polls the buffer for a matching message. If a matching message exists, it is correlated to the workflow instance. For a catch event, if multiple messages match the correlation information then the first published message is correlated.
+
+In the case of boundary events, a process does not need to be waiting for the message. In the case of a boundary event, the arrival of a correlated message will either cause an interruption to the current token flow (_interrupting boundary event_), or spawn a new token (in the case of a _non-interrupting boundary event_).
```

Comment:
This doesn't really belong to the message buffering. I think it is already covered by https://docs.zeebe.io/bpmn-workflows/message-events.html.